ui/apps/platform/cypress/integration/collections/collectionsCrudWorkflow.test.js

```diff
@@ -0,0 +1,184 @@
+
+        cy.get('button[aria-label="Add deployment label value for rule 1"]').click();
+        cy.get('input[aria-label="Select label value 3 of 3 for deployment rule 1 of 2"]').type(
+            'discover.*'
+        );
+        cy.get(`button:contains('Add "discover.*"')`).click();
+
+        cy.get(`button[aria-label='Delete mastercard.*']`).click();
+
+        cy.get('button[aria-label="Add cluster name value"]').click();
+        cy.get('input[aria-label="Select value 2 of 2 for the cluster name"]').type('staging');
+        cy.get(`button:contains('Add "staging"')`).click();
+
+        cy.get('input[aria-label="Select value 1 of 2 for the cluster name"]').type(
+            '{selectAll}security'
+        );
+        cy.get(`button:contains('Add "security"')`).click();
+
+        // Save
+        cy.get('button:contains("Save")').click();
+
+        // Revisit the collection page and verify that the changes have stuck
+        cy.get('a:contains("Financial deployments")').click();
+
+        // Check "byLabel" inputs for deployment
+        cy.get(`input[aria-label^="Select label value"][value="visa.*"]`);
+        cy.get(`input[aria-label^="Select label value"][value="discover.*"]`);
+        cy.get(`input[aria-label^="Select label value"][value="mastercard.*"]`).should('not.exist');
+
+        // Check "byName" inputs for namespace
+        cy.get(`input[aria-label$="for the namespace name"][value="payments"]`);
+
+        // Check "byName" inputs for cluster
+        cy.get(`input[aria-label$="for the cluster name"][value="staging"]`);
+        cy.get(`input[aria-label$="for the cluster name"][value="security"]`);
+        cy.get(`input[aria-label$="for the cluster name"][value="production"]`).should('not.exist');
+    });
+
+    it('should allow cloning an existing collection', () => {
+        // Cleanup from potential previous test runs
+        tryDeleteCollection(clonedName);
+        visitCollections();
+
+        // Make changes to an existing collection
+        cy.get(`a:contains("${collectionName}")`).click();
+        cy.get(`button:contains("Actions")`).click();
+        cy.get(`button:contains("Clone collection")`).click();
+
+        // Clone it with the default values
+        cy.get(`input[name="name"][value="${clonedName}"]`);
+        cy.get('button:contains("Save")').click();
+
+        // Ensure both collections are available
+        cy.get(`td[data-label="Collection"] a:contains("${collectionName}")`);
+        cy.get(`td[data-label="Collection"] a:contains("${clonedName}")`);
+    });
+
+    it('should delete the previously created collections', () => {
+        // From the main table
+        visitCollections();
+
+        // Delete the clone first, since the `:contains()` selector cannot do an exact match
+        // Delete one from the main collection table
+        cy.get(`tr:has(a:contains("${clonedName}")) button[aria-label="Actions"]`).click();
```

Comment:
Just realized that the long awaited `:has` CSS selector that landed in Chrome v105 can be used in our e2e tests since the current version of Chrome that Cypress is using is 108.

IMO this can simplify some complicated selections that make sense from a user perspective but were difficult to implement in a single selector so that we can take advantage of Cypress retry mechanisms. This is especially helpful when working with tables.

From an end user view, I'd scan the table for the collection I want to delete by name, go to the menu button, then click it. This CSS selector effectively says "Find me a `tr` that has a descendant `a` containing the text `"${clonedName}"`, then get me a button within that `tr` that is labeled `"Actions"` ".

I'm not sure if there is another way to do this with a single selector without hard coding a `tr` index or making other assumptions about the DOM structure.

Reply:
Super! Please share this excellent example in next team meeting.

Cypress folder has 7 occurrences in 4 files, which might have been via jQuery:
* https://api.jquery.com/has-selector/
* https://developer.mozilla.org/en-US/docs/Web/CSS/:has

Reply:
Aww, I thought I discovered something. Chances are this is still just using the jQuery selector under the hood...

Still good to know we can use it though :)